### ISSUE TITLE
paddle2.0版本无法读取保存后的模型进行预测

### ISSUE THREAD

**Issue by weihongwei-zg** [PaddlePaddle/Paddle#27970]
paddle2.0版本无法读取保存后的模型进行预测。我将点击反欺诈预测比赛的基线fork运行，用一个模型进行预测，每一次得到的结果都是不一样的，项目中采用model.load读取保存的模型参数，可是如果不读取，也会随机初始化参数，无法加载模型成功。
项目fork地址：https://aistudio.baidu.com/aistudio/projectdetail/701621

**Comment by yongqiangma** [comment]
能给看下出错相关代码和出错log吗？

**Comment by weihongwei-zg** [comment]
我代码公开了，您直接来运行fork，最底下有比较2次预测结果的情况。用同一个模型预测出来的都不一样，都是随机的。运行情况也在。https://aistudio.baidu.com/aistudio/projectdetail/1108352








------------------&nbsp;原始邮件&nbsp;------------------
发件人: "yongqiangma"<notifications@github.com&gt;; 
发送时间: 2020年10月15日(星期四) 晚上6:39
收件人: "PaddlePaddle/Paddle"<Paddle@noreply.github.com&gt;; 
抄送: "寒冰"<2426823455@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970)





 
能给看下出错相关代码和出错log吗？

**Comment by yongqiangma** [comment]
在 ai studio QQ群里问下吧。 群号 580959619

**Comment by guoshengCS** [comment]
可以先确认下参数是否被按照预期加载了，从model.network中取出参数打印看下两次是否一致。

**Comment by sunny0315** [comment]
也发现这个问题，请问有方法解决吗？

**Comment by weihongwei-zg** [comment]
不要用那种model.load方法读取。用rc1文档里面的方法读取



---原始邮件---
发件人: "kenney-fans"<notifications@github.com&gt;
发送时间: 2021年1月20日(周三) 中午11:10
收件人: "PaddlePaddle/Paddle"<Paddle@noreply.github.com&gt;;
抄送: "wobushihuair"<2426823455@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题: Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970)




 
也发现这个问题，请问有方法解决吗？

**Comment by sunny0315** [comment]
> 不要用那种model.load方法读取。用rc1文档里面的方法读取
> […](#)
> ---原始邮件--- 发件人: "kenney-fans"<notifications@github.com&gt; 发送时间: 2021年1月20日(周三) 中午11:10 收件人: "PaddlePaddle/Paddle"<Paddle@noreply.github.com&gt;; 抄送: "wobushihuair"<2426823455@qq.com&gt;;"Author"<author@noreply.github.com&gt;; 主题: Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970) 也发现这个问题，请问有方法解决吗？ — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

谢谢，可以发下参考链接吗

**Comment by weihongwei-zg** [comment]
https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-rc1/guides/02_paddle2.0_develop/08_model_save_load_cn.html------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;kenney-fans&quot;<notifications@github.com&gt;
发送时间:&nbsp;2021年1月20日(星期三) 中午11:14
收件人:&nbsp;&quot;PaddlePaddle/Paddle&quot;<Paddle@noreply.github.com&gt;;
抄送:&nbsp;&quot;wobushihuair&quot;<2426823455@qq.com&gt;;&quot;Author&quot;<author@noreply.github.com&gt;;
主题:&nbsp;Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970)

**Comment by sunny0315** [comment]
> [https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-rc1/guides/02_paddle2.0_develop/08_model_save_load_cn.html------------------&nbsp;原始邮件&nbsp;------------------](https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-rc1/guides/02_paddle2.0_develop/08_model_save_load_cn.html------------------&nbsp;%E5%8E%9F%E5%A7%8B%E9%82%AE%E4%BB%B6&nbsp;------------------) 发件人:&nbsp;&quot;kenney-fans&quot;<notifications@github.com&gt; 发送时间:&nbsp;2021年1月20日(星期三) 中午11:14 收件人:&nbsp;&quot;PaddlePaddle/Paddle&quot;<Paddle@noreply.github.com&gt;; 抄送:&nbsp;&quot;wobushihuair&quot;<2426823455@qq.com&gt;;&quot;Author&quot;<author@noreply.github.com&gt;; 主题:&nbsp;Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970)

参考这个文档，按下面的代码试了下还是不行（每次结果不一样，且与直接训练完infer的结果也不同），求指教！

# load and infer
layer_state_dict = paddle.load("work/model.pdparams")
opt_state_dict = paddle.load("work/model.pdopt")
model = SampleNet(TAGS, get_size_dict())
model.set_state_dict(layer_state_dict)
optimizer = paddle.optimizer.Adam(learning_rate=LEARNING_RATE, parameters=model.parameters())
optimizer.set_state_dict(opt_state_dict)

model = paddle.Model(model, inputs=inputs, labels=labels)

infer_reader = Reader(is_infer=True)
model.prepare()
infer_output = model.predict(infer_reader, TEST_BATCH_SIZE)
# 获取原始表中的字段并添加推理结果
result_df = infer_reader.df.loc[:, "sid"]
pack = []
for batch_out in infer_output[0]:
    for sample in batch_out:
        pack.append(np.argmax(sample))
# 保存csv文件
result_df = pd.DataFrame({"sid": np.array(result_df, dtype="int64"), "label": pack})
result_df.to_csv(RESULT_FILE, index=False)
print("结果文件保存至：", RESULT_FILE)

**Comment by weihongwei-zg** [comment]
我这个项目后面没有继续弄了，不过我用这些读取模型的方法在其他2.0rc1项目里面是正常的，就好奇怪。------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;&quot;kenney-fans&quot;<notifications@github.com&gt;
发送时间:&nbsp;2021年1月20日(星期三) 下午3:37
收件人:&nbsp;&quot;PaddlePaddle/Paddle&quot;<Paddle@noreply.github.com&gt;;
抄送:&nbsp;&quot;wobushihuair&quot;<2426823455@qq.com&gt;;&quot;Author&quot;<author@noreply.github.com&gt;;
主题:&nbsp;Re: [PaddlePaddle/Paddle] paddle2.0版本无法读取保存后的模型进行预测 (#27970)
